### PR TITLE
Transport review

### PR DESCRIPTION
**Pull request by SamJaarsma**
General review of the Transport Message Type, adding Baggage types according to IATA codelists and including segment-id for future SeatMap Message Type

### CODE REVIEW

**Comment by havan** [comment]
@SamJaarsma PRs should be made against `dev` branch.

**Comment by SamJaarsma** [comment]
Ahhh shit...


Sent from a mobile device, so please excuse any typos and formalities we would normally expect in written communications.

Many thanks and kind regards,

 [signatureImage]

Sam Jaarsma
VP Standards

T

:

+34 660 594305<tel:+34660594305>

M

:

***@***.******@***.***>

W

:

https://www.chain4travel.com<https://urldefense.com/v3/__https:/www.chain4travel.com__;!!CXsS5Xclluo!g6XfqLnXIkwKialVTSwVK5yi5Wygb4CfWFe4wDoPOeavZIN6bsLhsXa_pTCzGv6Uz_Az1HsG6gPgPbbDnLWbnmPh9C0S6FxlEa8bvA$>

________________________________

File: proto/cmp/types/v1alpha1/baggage.proto [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  // Dimensions
+  //
+  // Dimensions unit specification for length, width and height (cm or inches)
+  DimensionType dimension_type = 5;
+
+  // maximum height of the specified bagage type allowed
+  // maximum length, width and height of the specified bagage type allowed
+  int32 max_baggage_length = 6;
+  int32 max_baggage_width = 7;
+  int32 max_baggage_height = 8;
```
Comment (havan) [inline proto/cmp/types/v1alpha1/baggage.proto:30]: We should make this a message type of it's own. See next comment.

File: proto/cmp/types/v1alpha1/baggage.proto [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  BAGGAGE_TYPE_GOLF = 7; // Golf bags
+  BAGGAGE_TYPE_SURF = 8; // (wind)Surf or Kite boards
+}
+
+// Baggage dimensions unit ENUM type
+enum DimensionType {
+  DIMENSION_TYPE_UNSPECIFIED = 0; // No Baggage dimensions unit specified
+  DIMENSION_TYPE_CM = 1; // Centimeters
+  DIMENSION_TYPE_INCH = 2; // Inches
+}
```
Comment (havan) [inline proto/cmp/types/v1alpha1/baggage.proto:51]: We already have this in `distance.proto` file as `DistanceUnit`. We can create a `dimension.proto` and a message type like:

```protobuf
message Dimension {
  int32 lenght = 1;
  int32 width = 2;
  int32 height = 3;
  DistanceUnit unit = 4;
}
```

**Review by havan — CHANGES_REQUESTED** [review]

File: proto/cmp/types/v1alpha1/baggage.proto (hunk 2 above)
Comment (SamJaarsma) [reply proto/cmp/types/v1alpha1/baggage.proto:51]: we decided to implement more generalized measurements in another PR

File: proto/cmp/services/transport/v1alpha1/search_result_types.proto [hunk 3]
```diff
@@ -24,40 +24,41 @@ message TransportSearchResult {
   // Option to carry through an offer identifier for other stateful provider systems
   // like NDC AirShopping RS
   string offer_id = 2;
+  string segment_id = 3;
```
Comment (havan) [inline proto/cmp/services/transport/v1alpha1/search_result_types.proto]: This we already have under `TripSegment` message

**Review by havan — APPROVED** [review]

File: proto/cmp/services/transport/v1alpha1/search_result_types.proto (hunk 3 above)
Comment (SamJaarsma) [reply proto/cmp/services/transport/v1alpha1/search_result_types.proto]: corrected

**Pull request merged**